exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -746,79 +834,135 @@ func (p *Poloniex) GetOrderInfo(ctx context.Context, orderID string, pair curren
-				orderInfo.Status = order.Closed
-			}
-			return &orderInfo, nil
+		smartOrders, err = p.GetSmartOrderDetail(ctx, orderID, "")
+		if err != nil {
+			return nil, err
+		} else if len(smartOrders) == 0 {
+			return nil, order.ErrOrderNotFound
 		}
-		return nil, err
+		smartOrder = true
 	}
 
-	if orderInfo.Status, err = order.StringToOrderStatus(resp.Status); err != nil {
-		log.Errorf(log.ExchangeSys, "%s %v", p.Name, err)
+	var dPair currency.Pair
+	var oStatus order.Status
+	if smartOrder {
+		dPair, err = currency.NewPairFromString(smartOrders[0].Symbol)
+		if err != nil {
+			return nil, err
+		} else if !pair.IsEmpty() && !dPair.Equal(pair) {
+			return nil, fmt.Errorf("order with ID %s expected a symbol %v, but got %v", orderID, pair, dPair)
+		}
+		oType, err = order.StringToOrderType(smartOrders[0].Type)
+		if err != nil {
+			return nil, err
+		}
+		oStatus, err = order.StringToOrderStatus(smartOrders[0].State)
+		if err != nil {
+			return nil, err
+		}
+		oSide, err = order.StringToOrderSide(smartOrders[0].Side)
+		if err != nil {
+			return nil, err
+		}
+		return &order.Detail{
+			Price:         smartOrders[0].Price.Float64(),
+			Amount:        smartOrders[0].Quantity.Float64(),
+			QuoteAmount:   smartOrders[0].Amount.Float64(),
+			Exchange:      p.Name,
+			OrderID:       smartOrders[0].ID,
+			ClientOrderID: smartOrders[0].ClientOrderID,
+			Type:          oType,
+			Side:          oSide,
+			Status:        oStatus,
+			AssetType:     stringToAccountType(smartOrders[0].AccountType),
+			Date:          smartOrders[0].CreateTime.Time(),
+			LastUpdated:   smartOrders[0].UpdateTime.Time(),
+			Pair:          dPair,
+			Trades:        orderTrades,
+		}, nil
 	}
-	orderInfo.Price = resp.Rate
-	orderInfo.Amount = resp.Amount
-	orderInfo.Cost = resp.Total
-	orderInfo.Fee = resp.Fee
-	orderInfo.QuoteAmount = resp.StartingAmount
-
-	orderInfo.Side, err = order.StringToOrderSide(resp.Type)
+	dPair, err = currency.NewPairFromString(resp.Symbol)
 	if err != nil {
 		return nil, err
+	} else if !pair.IsEmpty() && !dPair.Equal(pair) {
+		return nil, fmt.Errorf("order with ID %s expected a symbol %v, but got %v", orderID, pair, dPair)
 	}
-
-	orderInfo.Date, err = time.Parse(time.DateTime, resp.Date)
+	oType, err = order.StringToOrderType(resp.Type)
 	if err != nil {
 		return nil, err
 	}
-
-	return &orderInfo, nil
+	oStatus, err = order.StringToOrderStatus(resp.State)
+	if err != nil {
+		return nil, err
+	}
+	oSide, err = order.StringToOrderSide(resp.Side)
+	if err != nil {
+		return nil, err
+	}
+	return &order.Detail{
+		Price:                resp.Price.Float64(),
+		Amount:               resp.Quantity.Float64(),
+		AverageExecutedPrice: resp.AvgPrice.Float64(),
+		QuoteAmount:          resp.Amount.Float64(),
+		ExecutedAmount:       resp.FilledQuantity.Float64(),
+		RemainingAmount:      resp.Quantity.Float64() - resp.FilledAmount.Float64(),
+		Cost:                 resp.Quantity.Float64() * resp.AvgPrice.Float64(),
```

Comment:
`resp.Quantity.Float64()` -> `resp.FilledQuantity.Float64()` maybe there is a partial fill?